sensors/simsensor_common/simsensor_reader.c

```diff
@@ -0,0 +1,297 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Simsensor reader reads and parses data form special scenario file
+ * than it pushes it to event_queue
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Piotr Nieciecki
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <ctype.h>
+#include <stdbool.h>
+
+#include "simsensor_reader.h"
+
+#define READER_END_SCENARIO_INDICATOR NO_SENSOR
+#define READER_DATA_SEPARATOR         ','
+
+#define READER_LINE_BUF_INIT_LEN 50
+
+#define READER_TIMESTAMP_NOT_SET -1
+
+
+typedef struct {
+	time_t firstTimestamp;
+	time_t lastTimestamp;
+	int loops;
+} simsens_time_t;
+
+
+struct simsens_reader_t {
+	FILE *scenario_file;
+	int sensor_types;
+
+	time_t time_horizon;
+
+	char *lineBuf; /* Buffer for storing line from file */
+	size_t bufLen;
+
+	simsens_time_t time_struct;
+};
+
+
+static int reader_skipHeader(simsens_reader_t *rd)
+{
+	size_t lineLen = getline(&rd->lineBuf, &rd->bufLen, rd->scenario_file);
+	if (lineLen < 1) {
+		return -1;
+	}
+
+	/* Checking if current line is header */
+	if (rd->lineBuf[0] == 's' || rd->lineBuf[0] == 'S') {
+		return 0;
+	}
+
+	return fseek(rd->scenario_file, 0, SEEK_SET);
```

Comment:
Is this function suppose to skip the header or only assert it? Currently on success of header check we will go back to the file start, possibly before the header.

Reply:
It suppose to skip the header. It gets whole line and checks if it is a header line.
If it is - just returns, so the next read, which uses this file descriptor gets data from the next line. 
In other case goes to the begging of the file.

Reply:
Ok, I have misunderstood it. Looks good.